tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+setting up a new webserver, we might reasonably want to increase the `LogLevel`
+from "warn" to "debug" to get more debugging messages, so let's run this
+command and use `sed` to make that change in the conffile:
+
+```bash
+sudo sed -e 's/LogLevel warn/LogLevel debug/' -i /etc/apache2/apache2.conf
+```
+
+Then, we'll restart our Apache2 server so that we can activate our configuration
+changes:
+
+```bash
+sudo systemctl restart apache2
+```
+
+Now if we run the `md5sum` command again, we can see the hash changed:
+
+```bash
+ubuntu@tutorial:~$ md5sum /etc/apache2/apache2.conf
+
+1109a77001754a836fb4a1378f740702  /etc/apache2/apache2.conf
+```
+
+This works great if we know that there's a file *we* changed, but what about if
+someone else tampered with a file, and we don't know which one? In that case,
+we can use:
+
+```bash
+dpkg --verify apache2
+```
+
+This will verify the checksums of the files on our system against those held in
+the package index for `apache2`, and return a rather strange looking result
+if (or when) it finds a mismatch:
+
+```text
+??5?????? c /etc/apache2/apache2.conf
+```
+
+Which is exactly what we were expecting to see, since we know we changed this
+file.
+
+But what if something else was messed with...something that shouldn't be, and
+something not changed by us? Let's make a "silly" change to a different file to
+test this -- in this case, changing all instances of the word "warning" to
+"silly" in a random package file:
+
+```bash
+sudo sed -e 's/warning/silly/' -i /usr/sbin/a2enmod
+```
+
+And then run the verification again with:
+
+```bash
+dpkg --verify apache2
+```
+
+We now see something that looks like this:
+
+```bash
+??5?????? c /etc/apache2/apache2.conf
+??5??????   /usr/sbin/a2enmod
+```
+
+```{note}
+You might have noticed there's a "c" next to the top line but not the
+bottom -- the "c" shows the file is a conffile.
+```
+
+`dpkg` can tell that the file has been changed, but won't tell us what the
+change was. However, since the file in question is not a conffile, we know that
+the change *won't be preserved* if we upgrade the package. This means that we
+can overwrite the changes and restore the default package content by
+"reinstalling" Apache2:
+
+```bash
+sudo apt install --reinstall apache2
+```
+
+By using the `--reinstall` flag, we can force `apt` to re-unpack all of the
+default content. If we then verify once more...
+
+```bash
+dpkg --verify apache2
+```
+
+...we can see that our change to the conffile has been preserved because
+`dpkg` the checksums are different, but the `a2enmod` file isn't
+listed anymore because it has been restored to the default. Phew!
+
+```{note}
+We can use `sudo apt install <package>` to upgrade an installed package, but
+this will only upgrade to the latest version. In our case, we were already on
+the latest version of Apache2, so we needed to force APT to re-unpack the
+content to overwrite our "silly" changes.
+```
+
+## Removing packages
```

Comment:
I like this, but I think you should have a little bit of this at the top in case the user accidentally installs something and gets off track from your tutorial. You could still have this section here but reword it slightly